### PR TITLE
5 split 1

### PR DESCRIPTION
**Pull request by cr3mebrulee**

### CODE REVIEW

**Comment by ioannnikulin** [comment]
1) ft_split_buf definitely looks like something that belongs to libft, not libft tests. So it probably should be declared in libft/sources/strings/strings.h and defined in libft/sources/strings/ft_split_buf.c

2) as for all these structures:
a) make sure they're called t_
b) if you use them in the function ft_split_buf and the consumer of the function does NOT need it, they should be declared in libft/sources/strings/strings_internal.h
c) if you use them in the function ft_split_buf and the consumer of the function does NOT need it AND you use them in tests, they should be declared in libft/sources/strings/strings_internal.h and then included from ft_split_buf_test.c (#include "../../sources/strings/strings_internal.h" or something)
d) if you use them in the function ft_split_buf and the consumer of the function DOES need them (probably not, hopefully not), they should be declared in libft/sources/strings/strings.h (then they'll leak into libft.h automatically) 
e) if you DO NOT use them in the function ft_split_buf, only in its tests, and the consumer of the function DOES NOT need them, they should be declared in libft/tests/strings/strings_tests.h

2) (following 1.e) I'd prefer to have only COMMON stuff in tests_common.h, like include libft, include assert, maybe some functions that are used in several testing components. If something is used in only ONE testing component though, technically it makes sense to put it in that tests's own libft/tests/tetedcomponent/testedcomponent_internal.h

3) we're not submitting any tests folders anyway. So if you need something only for tests, technically you can also declare and define that right in the testedcomponent_test.c file...

File: libft/sources/strings/ft_strjoin.c [hunk 1]
```diff
@@ -6,10 +6,11 @@
 /*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/11/13 19:01:40 by inikulin          #+#    #+#             */
-/*   Updated: 2024/09/30 14:40:29 by taretiuk         ###   ########.fr       */
+/*   Updated: 2024/09/30 17:27:26 by taretiuk         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
+
```
Comment (ioannnikulin) [inline libft/sources/strings/ft_strjoin.c:13]: Does Norme allow two consecutive empty lines?

File: Makefile [hunk 2]
```diff
@@ -22,12 +22,12 @@ OBJS = $(SRC_SRCS:.c=.o)
 ENDPOINT_OBJ = $(ENDPOINT_SRC:.c=.o)
 INCLUDES = -I . -I libft
 
-TEST_NAMES = main_test.c input_to_text_tree_test.c
+TEST_NAMES = main_test.c input_to_text_tree_test.c ft_split_buf_test.c ft_split_buf.c
```
Comment (ioannnikulin) [inline Makefile:25]: ft_split_buf.c goes to libft sources rule, not to test rule

File: tests/ft_split_buf.c [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	if (!res[*cwi][0])
+	{
+		free(res[*cwi]);
+		(*cwi)--;
+	}
+	return (0);
+}
+
+// Function to count the number of words based on the operators
+char	**ft_split_buf(const char *str, operator_array *op_array, int *sz)
```
Comment (ioannnikulin) [inline tests/ft_split_buf.c:112]: this function should be in libft/sources folder, also normed. I didn't norme the tests folder, but the sources - should be.

File: tests/ft_split_buf_test.c [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/09/21 11:13:18 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/09/30 10:49:49 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "tests_common.h"
+
+// static int	test_2(char **str_1, char **str_2, int sz_1)
```
Comment (ioannnikulin) [inline tests/ft_split_buf_test.c:15]: please remove the commented stuff once you're done with the whole file

File: libft/Makefile [hunk 5]
```diff
@@ -92,7 +92,7 @@ MEM_TEST_SRCS = $(addprefix $(TESTF)/$(MEM_F)/, $(MEM_TEST_NAMES))
 NUM_TEST_NAMES = ft_atoi_test.c ft_itoa_test.c ft_lltoa_base_test.c ft_ulltoa_base_test.c numbers_test.c
 NUM_TEST_SRCS = $(addprefix $(TESTF)/$(NUM_F)/, $(NUM_TEST_NAMES))
 
-STR_TEST_NAMES = ft_split.c ft_strlen.c ft_strchr.c ft_strdup.c ft_striteri.c ft_strjoin.c ft_strlcpy.c ft_strlcat.c ft_strmapi.c ft_strncmp.c ft_strnstr.c ft_strrchr.c ft_strtrim.c ft_substr.c ft_putchar_fd.c ft_putendl_fd.c ft_putnbr_fd.c ft_putstr_fd.c
+STR_TEST_NAMES = ft_split.c ft_strlen.c ft_strchr.c ft_strdup.c ft_striteri.c ft_strjoin.c ft_strlcpy.c ft_strlcat.c ft_strmapi.c ft_strncmp.c ft_strnstr.c ft_strrchr.c ft_strtrim.c ft_substr.c ft_putchar_fd.c ft_putendl_fd.c ft_putnbr_fd.c ft_putstr_fd.c ft_split_buf_test.c
 STR_TEST_SRCS = $(addprefix $(TESTF)/$(STR_F)/, $(subst .c,_test.c,$(STR_TEST_NAMES)) strings_test.c)
```
Comment (ioannnikulin) [inline libft/Makefile:96]: Here a `_test.c` suffix is added automatically to every filename. If you read the compilation log carefully, you could have noticed your error says `ft_split_buf_test_test.c` ;)

**Comment by cr3mebrulee** [comment]
> 1. ft_split_buf definitely looks like something that belongs to libft, not libft tests. So it probably should be declared in libft/sources/strings/strings.h and defined in libft/sources/strings/ft_split_buf.c
> 
>     2. as for all these structures:
>        a) make sure they're called t_
>        b) if you use them in the function ft_split_buf and the consumer of the function does NOT need it, they should be declared in libft/sources/strings/strings_internal.h
>        c) if you use them in the function ft_split_buf and the consumer of the function does NOT need it AND you use them in tests, they should be declared in libft/sources/strings/strings_internal.h and then included from ft_split_buf_test.c (#include "../../sources/strings/strings_internal.h" or something)
>        d) if you use them in the function ft_split_buf and the consumer of the function DOES need them (probably not, hopefully not), they should be declared in libft/sources/strings/strings.h (then they'll leak into libft.h automatically)
>        e) if you DO NOT use them in the function ft_split_buf, only in its tests, and the consumer of the function DOES NOT need them, they should be declared in libft/tests/strings/strings_tests.h
> 
>     3. (following 2.e) I'd prefer to have only COMMON stuff in tests_common.h, like include libft, include assert, maybe some functions that are used in several testing components. If something is used in only ONE testing component though, technically it makes sense to put it in that tests's own libft/tests/tetedcomponent/testedcomponent_internal.h
> 
>     4. we're not submitting any tests folders anyway. So if you need something only for tests, technically you can also declare and define that right in the testedcomponent_test.c file...

@ioannnikulin i guess if i declare the prototype on ft_split_buf() with structure in parameters - i make it by way that consumer needs them? :-| (If yes, maybe for now i just keep it like this and think how to change it later?...)

File: libft/sources/strings/ft_strjoin.c (hunk 1 above)
Comment (cr3mebrulee) [reply libft/sources/strings/ft_strjoin.c:13]: No, there is no norm yet

File: libft/Makefile (hunk 5 above)
Comment (cr3mebrulee) [reply libft/Makefile:96]: ok, thanks for hint! ^_^

**Comment by ioannnikulin** [comment]
@cr3mebrulee nope, the user would still need to instantiate an array of those, so he needs access to their definitions

**Pull request closed without merging**